polars/polars-core/src/chunked_array/mod.rs

```diff
@@ -551,18 +532,18 @@ where
             DataType::Int64 => downcast_and_pack!(Int64Array, Int64),
             DataType::Float32 => downcast_and_pack!(Float32Array, Float32),
             DataType::Float64 => downcast_and_pack!(Float64Array, Float64),
-            DataType::Date32 => downcast_and_pack!(Date32Array, Date32),
-            DataType::Date64 => downcast_and_pack!(Date64Array, Date64),
+            DataType::Date32 => downcast_and_pack!(Int32Array, Date32),
+            DataType::Date64 => downcast_and_pack!(Int64Array, Date64),
             DataType::Time64(TimeUnit::Nanosecond) => {
-                let v = downcast!(Time64NanosecondArray);
```

Comment:
in arrow2, downcast only needs to happen for physical types. All logical variations have the same physical type and thus do not have corresponding primitive arrays. This is an example of such change, whereby we downcast the array to the corresponding physical type.

Reply:
So, it is safe to transmute to the same physical type as the logical type? Given that we correctly transmute of course. 

This may reduce a lot of code branches and generic monomorphizations. :)

Reply:
All is safe if we use `as_any().downcast_ref().unwrap`, as it panics otherwise, but yes, every logical type has a unique physical type backing it. We have a table with the exact map [in the guide](https://jorgecarleitao.github.io/arrow2/high_level.html).